common/environment/environment.cpp

```diff
@@ -1058,6 +1126,75 @@ IConstMachineInfo * CLocalEnvironment::getMachineByAddress(const char * name) co
     return (CConstMachineInfo *) cached;
 }
 
+IConstMachineInfo * CLocalEnvironment::getMachineForLocalHost() const
+{
+    IConstEnvBase *cached = getCache("Hardware/Computer[@netAddress='.']");
+    if (!cached)
+    {
+        StringBuffer localHostIP;
+        queryHostIP().getIpText(localHostIP);
+
+        StringArray possibleIPs;
+        possibleIPs.append(".");
+        possibleIPs.append("localhost");
+        possibleIPs.append("127.0.0.1");
+        possibleIPs.append(localHostIP.str());
+
+        synchronized procedure(safeCache);
+        ForEachItemIn(i, possibleIPs)
+        {
+            VStringBuffer xpath("Hardware/Computer[@netAddress='%s']", possibleIPs.item(i));
+            IPropertyTree *d = p->queryPropTree(xpath.str());
+            if (d)
+            {
+                cached = new CConstMachineInfo((CLocalEnvironment *) this, d);
+                setCache("Hardware/Computer[@netAddress='.']", cached);
+                break;
+            }
+        }
+    }
+    return (CConstMachineInfo *) cached;
+}
+
+IConstDropZoneInfo * CLocalEnvironment::getDropZone(const char * name) const
+{
+    if (!name)
+        return NULL;
+    buildDropZoneCache();
+    VStringBuffer xpath("Software/DropZone[@name=\"%s\"]", name);
+    synchronized procedure(safeCache);
+    IConstEnvBase *cached = getCache(xpath.str());
+    if (!cached)
```

Comment:
I wrote this method based on CLocalEnvironment::getMachine(). You are correct. If the name does not exist as a valid DropZone, it will return NULL even if I try one more time.